src/NServiceBus.Core/DelayedDelivery/TimeoutManager/FailureInfoStorage.cs

```diff
@@ -13,7 +13,7 @@ public FailureInfoStorage(int maxElements = 1000)
             this.maxElements = maxElements;
         }
 
-        public void RecordFailureInfoForMessage(string messageId, Exception exception)
+        public void RecordFailureInfoForMessage(string messageId, Exception exception, bool shouldMoveToErrorQueue = false, bool shouldDeferForRetry = false)
```

Comment:
Instead of adding flags in signature can we have separate methods that `SLR` and `Errors` would use? For example `MarkForSLRDeferral` and `MarkForMovingToErrorQueue`.